plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -89,6 +97,42 @@ export class TeamManager {
         }
     }
 
+    public async fetchBillingUsage(organization_id: string): Promise<BillingUsage | null> {
+        const cachedUsage = this.billingUsageCache.get(organization_id)
+        if (cachedUsage !== undefined) {
+            return cachedUsage
+        }
+
+        const timeout = timeoutGuard(`Still running "fetchBillingUsage". Timeout warning after 30 sec!`)
+        try {
+            let usage: BillingUsage | null = await this.db.fetchBillingUsage(organization_id)
+            if (!usage) {
+                usage = {}
+            }
+            this.billingUsageCache.set(organization_id, usage)
+            return usage
+        } finally {
+            clearTimeout(timeout)
+        }
+    }
+
+    public async isAboveUsageLimit(organization_id: string, key: string): Promise<boolean> {
```

Comment:
This needs unit tests to make sure _all_ implicit behavior here is documented (e.g. what happens if postgres query fails)?